### PR TITLE
Initialize criterion

### LINKED ISSUE #333: makeMBOInfillCriterionCB lambda default
@berndbischl and I agreed, that at the moment the style of `makeMBOInfillCriterionCB` is not very good. This function is called quite often and each time the lambda value is again determined. 
Intstead it should be fixed once the par.set ist known and then probably also accessible through `$params$cb.lambda`.

### PR DESCRIPTION
**Pull request by ja-thomas**
Fixes #333 and makes PR #340 obsolete

Some comments:

- New subclass for infill crits is now required, not sure about the naming. Is `InfillCritcb`, `InfillCritei` etc. ok? Other suggestions are welcome :)

- `isSimpleNumeric` should probably go into PH

- I'm actually confused why `test_infillcrits` did not fail as we have fully numeric function but checked against `cb.lambda = 1`. After the refactoring it actually failed (as it should have) so I changed it.

- Do we want to export `initializeCriterion` ?

- The fix in `proposePointsDIB` is not so nice, but we want to change the code there anyways, right?

### CODE REVIEW

**Comment by berndbischl** [comment]
- lets please short names, if they contain the same info. minimum desciption length principle
so: initCriterion. actually initCrit would better. can we rename "Criterion" everywhere? but then in a separate issue.

- names: InfillCritEI, InfilCritCB

- initializeCriterion: i think you need to export this so people can implement new methods from the outside

- please open issue about isSimpleNumeric and copy it to PH now

- dont forget to check that we have an issue for the DIB problems discussed in #340 

otherwise looks good. @jakobbossek ?

**Comment by ja-thomas** [comment]
> lets please short names, if they contain the same info. minimum desciption length principle
> so: initCriterion. actually initCrit would better. can we rename "Criterion" everywhere? but then in a separate issue.

Done and done #348

> 
> names: InfillCritEI, InfilCritCB

Done

> 
> initializeCriterion: i think you need to export this so people can implement new methods from the outside

Done

> 
> please open issue about isSimpleNumeric and copy it to PH now
> 

Will do

> dont forget to check that we have an issue for the DIB problems discussed in #340
> 

There is PR #346, I would like to merge this first. I had to touch the DIB only to make the tests pass. The real fix for this problem should be in #346 not here 

> otherwise looks good. @jakobbossek ?

**Comment by jakob-r** [comment]
> There is PR #346, I would like to merge this first. I had to touch the DIB only to make the tests pass. The real fix for this problem should be in #346 not here

With *this* you mean PR #346 or `this`? :thinking:

**Comment by ja-thomas** [comment]
> > There is PR #346, I would like to merge this first. I had to touch the DIB only to make the tests pass. The real fix for this problem should be in #346 not here

> With this you mean PR #346 or this?

`this`

#347 before #346 

(actually it doesn't matter, I just don't want to do both in one PR)

File: tests/testthat/test_infillcrits.R [hunk 1]
```diff
@@ -33,7 +33,7 @@ test_that("infill crits", {
       expect_true(!anyMissing(opdf$infill_ei[, c("ei","se","mean")]))
     if (!is.null(opdf$infill_cb)) {
       expect_true(!anyMissing(opdf$infill_cb[, c("se","mean","lambda")]))
-      expect_true(all(opdf$infill_cb$lambda == 2))
+      expect_true(all(opdf$infill_cb$lambda == 1))
```
Comment (jakob-r) [inline tests/testthat/test_infillcrits.R]: okay - you changed here that lambda should be 1 but in line 56 is specifically says `crit.cb2` which leads me to think that something is seriously not right here.

File: tests/testthat/test_initCrit.R [hunk 2]
```diff
@@ -0,0 +1,19 @@
+test_that("infill crit initialization", {
+
+  ps = makeParamSet(
+    makeNumericParam("x1", lower = -2, upper = 1),
+    makeIntegerParam("x2", lower = -1, upper = 2)
+  )
+  
+  expect_equal(crit.cb1, initCrit(crit.cb, ps))
```
Comment (jakob-r) [inline tests/testthat/test_initCrit.R]: as observed above. please add a test that checks that initCrit does not overwrite a specified criterion

File: tests/testthat/test_infillcrits.R (hunk 1 above)
Comment (ja-thomas) [inline tests/testthat/test_infillcrits.R]: Thanks for noticing, that is actually a pretty big bug :(

in `infillCrit.InfillCritCB` it should be `crit$params$cb.lambda` and not `crit$cb.lambda`

fixing now

**Comment by berndbischl** [comment]
1) note the hourlong disucussion about namechanges. maybe stick with criterion? i closed my request there.

2) i think the code is good now, but dont know the best merge order. you had lots of comments about this here

finalize quickly pls

**Comment by jakob-r** [comment]
After #348 was closed after a short discussion I will rename crit to criterion here.

**Comment by jakob-r** [comment]
I am a bit confused after #348. Should it be initCriterion now?

**Comment by ja-thomas** [comment]
> note the hourlong disucussion about namechanges. maybe stick with criterion? i closed my request there.

so, `initializeCriterion` and `infillCriterionCB` like it was in the beginntn, OK? (If nobody is objecting I'm doing it)

> 
> i think the code is good now, but dont know the best merge order. you had lots of comments about this here

This first? It shouldn't touch too much of the other PR's

> 
> finalize quickly pls

**Pull request merged**